pkg/mountutil/volumestore/volumestore.go

```diff
@@ -83,13 +83,23 @@ func (vs *volumeStore) Create(name string, labels []string) (*native.Volume, err
 	}
 	volPath := filepath.Join(vs.dir, name)
 	volDataPath := filepath.Join(volPath, DataDirName)
-	fn := func() error {
+	fn := func() (err error) {
 		if err := os.Mkdir(volPath, 0700); err != nil {
 			return err
 		}
+		defer func() {
+			if err != nil {
+				os.Remove(volPath)
```

Comment:
> Volume path should be cleaned up if create volume failed.

cleanup doesn't seem to happen for errors outside of this fn?

Reply:
cleanup is called in this fn for the two errors, `json.MarshalIndent` and `os.WriteFile`
Creation is done after this fn  is completed. No other new errors outside of it. If some action added after `lockutil.WithDirLock(vs.dir, fn)`  in future, cleanup will be moved out of this fn.

Reply:
> cleanup is called in this fn for the two errors, json.MarshalIndent and os.WriteFile

cleanup seems to fail if that dir isn't empty

>  in future, cleanup will be moved out of this fn.

should be documented

Reply:
> cleanup seems to fail if that dir isn't empty
dir is empty as expected if writing json file failed. Or use `os.RemoveAll` instead ? 

> should be documented
OK, should I add some comments？

Reply:
RemoveAll can be dangerous, so I don't prefer it

Reply:
> dir is empty as expected if writing json file failed. 

https://pkg.go.dev/os#WriteFile
>  Since WriteFile requires multiple system calls to complete, a failure mid-operation can leave the file in a partially written state.

Reply:
How about these steps in cleanup ?
step 1.  check json file exists and remove it
step 2. remove volume data path dir
step 3. remove volume path dir

Reply:
the path was created on multiple steps so I beleive that cleanup should follow the same pattern and be  cleaned on multiple steps

Reply:
> How about these steps in cleanup ?
> step 1. check json file exists and remove it

fixed
Use `stat` to check json file exist
Also add some comments.